kurtosis-devnet/pkg/deploy/template.go

```diff
@@ -20,20 +21,58 @@ type Templater struct {
 	dataFile     string
 	buildDir     string
 	urlBuilder   func(path ...string) string
+
+	// Common state across template functions
+	buildJobsMu sync.Mutex
```

Comment:
nit: mutexes are often abbreviated to `mux`, so this may be more readable as `buildJobsMux`, or `jobsMux` if you prefer shorter.